pyat/at/load/madx.py

```diff
@@ -475,9 +477,9 @@ def flatten(self, parser, offset: float = 0.0, refer: float = 1.0, refpos=None):
 
     def expand(self, parser: MadxParser) -> Generator[elt.Element, None, None]:
         def insert_drift(dl, el):
-            if dl > 1.0e-10:
+            if dl > 1.0e-5:
```

Comment:
Is there a reason to change this tolerance?

Reply:
From the MADX manual:

_When the sequence is expanded in a [USE](https://mad.web.cern.ch/mad/webguide/manual.html#Ch4.S10) command, MAD-X generates the drift spaces between elements according to the following rules:_

_1. When the distance between the exit of the previous element and the entrance of the next element is positive and greater than a threshold of **1μm** an explicit drit is generated with its own name, unless an already existing drift space with the same length (within 10−12m tolerance) can be re-used.
2. When the absolute value of the distance between the exit of the previous element and the entrance of the next element is less than a given tolerance of **1μm**, no drift space is created and the elements are considered as contiguous.Note that in very specific cases this can cause very small errors to accumulate and the actual length of the sequence can vary slightly from the declared length. (see second example below)
3. When the distance between the exit of the previous element and the entrance of the next element is negative and less than **−1μm**, the elements are considered to be overlapping and MAD-X terminates with a “negative drift length” fatal error._

MADX sets the threshold at 1.E-6. The previous value causes the generation of many extremely small drifts. 1.E-6 still caused a few unwanted drifts absent in MADX, so I increased to 1.E-5. I can put back 1.E-6 if you prefer, a few μm long drifts are harmless.